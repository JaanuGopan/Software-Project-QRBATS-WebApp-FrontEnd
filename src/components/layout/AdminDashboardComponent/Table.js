import React, { useState, useEffect } from "react";
import "../../../pages/AdminDashboard/AdminDashboard.css";
import { FaEdit } from "react-icons/fa";

const Table = ({ search, handleUpdateEvent, onEventClick, eventList }) => {
  const [events, setEvents] = useState([]);
  const [selectedEvent, setSelectedEvent] = useState(null);

  useEffect(() => {
    // Update events whenever eventList prop changes
    setEvents(eventList);
  }, [eventList]);

  const handleEventClick = (event) => {
    setSelectedEvent(event);
    onEventClick(event);
  };

  return (
    <div className="tableDesign">
      <table className="tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Name</th>
            <th>Start Date</th>
            <th>End Date</th>
            <th>Venue</th>
            <th>Start Time</th>
            <th>End Time</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {events
            .filter(
              (event) =>
                event.eventName.toLowerCase().includes(search.toLowerCase()) ||
                event.eventVenue.toLowerCase().includes(search.toLowerCase())
            )
            .map((event, index) => (
              <tr
                key={index}
                onClick={() => handleEventClick(event)}
                className={
                  selectedEvent === event ? "selected-row" : "event-row"
                }
              >
                <td>{index + 1}</td>
                <td>{event.eventName}</td>
                <td>{event.eventDate}</td>
                <td>{event.eventValidDate}</td>
                <td>{event.eventVenue}</td>
                <td>{event.eventTime}</td>
                <td>{event.eventEndTime}</td>
                <td>
                  <button
                    onClick={() => handleUpdateEvent(event)}
                    className="EditButton"
                  >
                    <FaEdit className="EditIcon" />
                  </button>
                </td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default Table;
