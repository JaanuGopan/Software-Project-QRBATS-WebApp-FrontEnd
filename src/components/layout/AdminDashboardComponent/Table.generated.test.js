import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import { FaEdit } from "react-icons/fa";
import Table from "./Table";

jest.mock("../../../pages/AdminDashboard/AdminDashboard.css");
jest.mock("react-icons/fa");

const renderTree = tree => renderer.create(tree);
describe('<Table>', () => {
  it('should render component', () => {
    expect(renderTree(<Table 
    />).toJSON()).toMatchSnapshot();
  });
  
});