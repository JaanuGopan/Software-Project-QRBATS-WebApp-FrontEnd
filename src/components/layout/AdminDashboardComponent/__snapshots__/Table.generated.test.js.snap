// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table> should render component 1`] = `
<div
  className="tableDesign"
>
  <table
    className="tableArrangement"
  >
    <thead>
      <tr>
        <th>
          No
        </th>
        <th
          className="expand"
        >
          Name
        </th>
        <th>
          Start Date
        </th>
        <th>
          End Date
        </th>
        <th>
          Venue
        </th>
        <th>
          Start Time
        </th>
        <th>
          End Time
        </th>
        <th>
          Edit
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
